addons/sale/models/sale_order.py

```diff
@@ -737,17 +736,31 @@ def action_quotation_send(self):
         }
 
     def _find_mail_template(self, force_confirmation_template=False):
-        template_id = False
+        """ Get the appropriate mail template for the current sales orders based on their state.
 
-        if force_confirmation_template or (self.state == 'sale' and not self.env.context.get('proforma', False)):
-            template_id = int(self.env['ir.config_parameter'].sudo().get_param('sale.default_confirmation_template'))
-            template_id = self.env['mail.template'].search([('id', '=', template_id)]).id
-            if not template_id:
-                template_id = self.env['ir.model.data']._xmlid_to_res_id('sale.mail_template_sale_confirmation', raise_if_not_found=False)
-        if not template_id:
-            template_id = self.env['ir.model.data']._xmlid_to_res_id('sale.email_template_edi_sale', raise_if_not_found=False)
+        If all SOs are confirmed or ``force_confirmation_template`` is `True`, we return the mail
+        template for the sale confirmation. Otherwise, we return the quotation email template.
 
-        return template_id
+        :param bool force_confirmation_template: Allows to force the confirmation template to be
+        used in any case
+        :return: The correct mail template based on the current status
+        :rtype: record of `mail.template` or `None` if not found
+        """
+        if force_confirmation_template or (
+            all(order.state == 'sale' for order in self) and not self.env.context.get('proforma')
+        ):
+            default_confirmation_template_id = self.env['ir.config_parameter'].sudo().get_param(
+                'sale.default_confirmation_template', 0
+            )
+            default_confirmation_template = self.env['mail.template'].browse(
+                default_confirmation_template_id
+            ).exists()
```

Comment:
The potentially unnecessary browse reported in https://github.com/odoo/odoo/pull/72368#discussion_r895490824 is still here.
```suggestion
            default_confirmation_template_id = self.env['ir.config_parameter'].sudo().get_param(
                'sale.default_confirmation_template'
            )
            default_confirmation_template = default_confirmation_template_id and self.env['mail.template'].browse(
                default_confirmation_template_id
            ).exists()
```